CheckRightsService_RemoveRightsFromUser

KostinP opened the pull request:
I started from https://github.com/LT-Students/DigitalOffice/pull/49 pull request so that there were fewer conflicts. Didn't make a repository, it doesn't work
(You also need to add controller tests)

KostinP committed:
Added base of solution of task "Add endpoint RemoveRightsFromUser"

denis-kriv commented on DigitalOffice/Services/CheckRightsService/Validator/RemoveRightsFromUserValidator.cs:
```diff
@@ -0,0 +1,19 @@
+{
+    public class RemoveRightsFromUserValidator : AbstractValidator<RemoveRightsFromUserRequest>
+    {
+        public RemoveRightsFromUserValidator()
+        {
+            RuleFor(rights => rights.UserId)
+                .NotEmpty()
+                .WithName("User Id");
+
+            RuleFor(rights => rights.RightsId)
+                .NotEmpty()
+                .WithName("Right Id");
+        }
```
I am not sure, but maybe we have more rules.

denis-kriv replied:
have to add*

denis-kriv commented on DigitalOffice/Services/CheckRightsService/Controllers/CheckRightsController.cs:
```diff
@@ -14,5 +14,13 @@ public List<Right> GetRightsList([FromServices] IGetRightsListCommand command)
             return command.Execute();
         }
+
+        [HttpPost("removeRightsFromUser")]
```
Why not to use `HttpDelete`

KostinP replied:
Fixed.

denis-kriv commented on DigitalOffice/Tests/CheckRightsServiceTests/Validators/RemoveRightsFromUserValidatorTests.cs:
```diff
@@ -0,0 +1,33 @@
+        private IValidator<RemoveRightsFromUserRequest> validator;
+
+        [SetUp]
+        public void Initialize()
```
`SetUp`

KostinP replied:
Fixed.

KostinP committed:
Mini fixes + attempt to fix a bug (failed)

KostinP committed:
Changed HttpPost to HttpDelete

DarkFelas commented on DigitalOffice/Services/CheckRightsService/Repositories/CheckRightsRepository.cs:
```diff
@@ -23,5 +25,37 @@ public List<Right> GetRightsList()
+                {
+                    // TODO: add custom exception
+                    throw new Exception("Right doesn't exist.");
+                }
```
remove

KostinP committed:
Update solution

- Added table RightUsers to CheckRightsServiceDbContext
- Remove Fluent API from DbRight
- Rename Migration addrightstodb to AddRightsToDb (see next commit)
- Rename RightsId to RightIds
- Update repository
- Update tests

KostinP committed:
Fix migrations

KostinP committed:
Added new migration

KostinP commented:
Problem: when creating a new migration, it still tries to delete data.
@mekk1t @DarkFelas 

Perhaps Nikita did not have this, because he did not touch the file CheckRightsServiceDbContextModelSnapshot.cs that is responsible for the current state of the model stored. It enables the migrations framework to calculate the changes required to bring the database up to date with the model.

What to do?

DarkFelas commented on DigitalOffice/Services/CheckRightsService/Commands/Interfaces/IRemoveRightsFromUserCommand.cs:
```diff
@@ -0,0 +1,18 @@
+        /// </summary>
+        /// <param name="request">Request with rights and user id.</param>
+        /// <returns>Return true if successfully else return false.</returns>
+        bool Execute(RemoveRightsFromUserRequest request);
```
no need for bool. just **void**

KostinP replied:
Fixed.

DarkFelas reviewed (approved):
Does not saw any problem with migration, all works well.

KostinP committed:
Merge branch 'develop' into feature/CheckRightsService_RemoveRightsFromUser

KostinP committed:
Resolve conflicts after merge

KostinP committed:
Remove HasData from Snapshot (now new migrations will not delete this data)

KostinP committed:
true -> void

KostinP committed:
Updated packages

KostinP committed:
bool -> void

KostinP committed:
Update test

KostinP committed:
Removed extra usings

KostinP committed:
Revert "Updated packages"

This reverts commit 8f1100d1e1d3c8340a4d8d0d2877d34c9e75f67a.

KostinP committed:
Update GetRightsListCommand.cs

lalagvana commented on Tests/CheckRightsServiceTests/Repositories/CheckRightsRepositoryTests.cs:
```diff
@@ -33,22 +75,96 @@ public void Clear()
+        }
+
+        [Test]
+        public void ShouldNotDeleteAnythingWhenRightIdIsNoFound()
```
IsNotFound

KostinP replied:
Fixed.

djino97 commented on Services/CheckRightsService/Commands/RemoveRightsFromUserCommand.cs:
```diff
@@ -0,0 +1,30 @@
+        {
+            validator.ValidateAndThrow(request);
+            repository.RemoveRightsFromUser(request);
+
+            // TODO: Check rights of user (sender) with Kernel_AccessValidator
+        }
```
Will you add check rights in the this pr?

KostinP replied:
Fixed.

mekk1t reviewed (approved):
Add AccessValidator

denis-kriv reviewed (approved):
Add some rules for validator please

KostinP committed:
Merge branch 'develop' into feature/CheckRightsService_RemoveRightsFromUser

KostinP committed:
Added AccessValidator

KostinP committed:
Fix typo
